runtime/compiler/runtime/Trampoline.cpp

```diff
@@ -480,25 +480,36 @@ bool ppcCodePatching(void *method, void *callSite, void *currentPC, void *curren
 
          oldBits = *(int32_t *)((uint8_t *)callSite - encodingStartOffset);     // The load of the first IPIC cache slot or rldimi
 #if defined(TR_TARGET_64BIT)
-         if (((oldBits>>26) & 0x0000003F) != 30)
+         if (TR::Compiler->target.cpu.isAtLeast(OMR_PROCESSOR_PPC_P10))
             {
-            // PTOC was used, oldBits is ld
-            currentDistance = oldBits<<16>>16;
-            if (((oldBits>>16) & 0x0000001F) == 12)
-               {
-               oldBits = *(int32_t *)((uint8_t *)callSite - encodingStartOffset - 4);
-               currentDistance += oldBits<<16;
-               }
-            patchAddr = *(uint8_t **)(*(intptr_t *)extra + currentDistance);
+            // oldBits is the latter half of paddi
+            distance = ((*(int32_t *)((uint8_t *)callSite - encodingStartOffset - 4)) & 0x0003FFFF) << 16;
```

Comment:
Should not it be shift left by 14, not 16 ?

Reply:
no, it should be shift left by 16.  the 34-bit offset is composed of  (top-18-bit << 16) |  (bottom-16-bit).  shift left by 14 will overlap two bits, and the whole offset will be 32bit at most.

your comment did remind me of the possibility of losing bits, if the shift is conducted on a 32-bit type.  so, i type-cast the 18-bit to intptr_t in order to not lose bits possibly.  in reality, it cannot happen though (since this particular offset must be positive and less than this particular method compiled length).  amend this commit as below.

Reply:
Done